test/stdlib/tgmath.swift.gyb

```diff
@@ -53,6 +53,18 @@ func expectEqualWithTolerance<T>(_ expected: TestLiteralType, _ actual: T,
              file: file, line: line)
 }
 
+func checkGradient<T: BinaryFloatingPoint & Differentiable>(
+  _ f: @differentiable (T, T) -> T,
+  _ x: T,
+  _ y: T) where T == T.TangentVector {
+  let eps = T(0.001)
+  let grad = gradient(at: x, y, in: f)
+  let dfdx = (f(x + eps, y) - f(x, y)) / eps
+  let dfdy = (f(x, y + eps) - f(x, y)) / eps
+  expectEqualWithTolerance(TestLiteralType(dfdx.rounded(.toNearestOrAwayFromZero)), grad.0)
+  expectEqualWithTolerance(TestLiteralType(dfdy.rounded(.toNearestOrAwayFromZero)), grad.1)
```

Comment:
Is the rounding necessary? Can you make this work by making it more tolerant instead of by rounding?

More tolerance seems like a safer test than rounding, because rounding could mask for example a situation where the gradient is supposed to be `1.8` but we calculate it as `2`. (Based on my understanding, the gradients will actually always be integers, so this hypothetical situation will never actually happen. But tests are good for catching cases where we have misunderstood something, so it would be good to have a test that could catch such a situation.)

Reply:
Given the `eps` value I was using ( 0.001)  I needed to use tolerance of around 3700 for the check to pass, which seemed high to me, hence I opted for rounding, but you make a good point on having the ability to catch future errors here.

If I increase the eps to be `0.1` then the test pass with a tolerance of 32.

done in 7ce2e4c